lib/std/fs/wasi.zig

```diff
@@ -22,6 +23,25 @@ pub const Preopens = struct {
         }
         return null;
     }
+
+    pub fn findDir(p: Preopens, full_path: []const u8, flags: std.fs.Dir.OpenDirOptions) std.fs.Dir.OpenError!std.fs.Dir {
+        if (p.names.len <= 2)
```

Comment:
Much better, but this will still open the preopen at index '0' when none of the preopens are matching to given `full_path`. Perhaps a check ater the loop that returns error.NotFound when `fd` is still `0`.

Reply:
right! I was probably assuming (wrongly) that there is always at least `/`...